packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionHierarchy_def.hpp

```diff
@@ -587,15 +590,23 @@ void createRegionHierarchy(const int maxRegPerProc,
   /* Get coarse level matrices and prolongators from MueLu hierarchy
    * Note: fine level has been dealt with previously, so we start at level 1 here.
    */
+  using real_type = typename Teuchos::ScalarTraits<Scalar>::coordinateType;
+  using realvaluedmultivector_type = Xpetra::MultiVector<real_type, LocalOrdinal, GlobalOrdinal, Node>;
+  Array<RCP<realvaluedmultivector_type> > regCoarseCoordinates(maxRegPerProc);
   for (int l = 1; l < numLevels; ++l) { // Note: we start at level 1 (which is the first coarse level)
     for (int j = 0; j < maxRegPerProc; ++j) {
       RCP<MueLu::Level> level = regGrpHierarchy[j]->GetLevel(l);
 
-      regProlong[l][j] = level->Get<RCP<Matrix> >("P", MueLu::NoFactory::get());
+      if(l == numLevels - 1) {
+        if(mapComp->getComm()->getRank() == 0) level->print(std::cout, MueLu::Extreme);
+        regCoarseCoordinates[j] = level->Get<RCP<realvaluedmultivector_type> >("Coordinates2", MueLu::NoFactory::get());
+      }
+
+      regProlong[l][j]  = level->Get<RCP<Matrix> >("P", MueLu::NoFactory::get());
       regMatrices[l][j] = level->Get<RCP<Matrix> >("A", MueLu::NoFactory::get());
 
-      regRowMaps[l][j] = Teuchos::rcp_const_cast<Xpetra::Map<LO,GO,NO> >(regMatrices[l][j]->getRowMap()); // ToDo (mayr.mt) Should we rather copy?
-      regColMaps[l][j] = Teuchos::rcp_const_cast<Xpetra::Map<LO,GO,NO> >(regMatrices[l][j]->getColMap()); // ToDo (mayr.mt) Should we rather copy?
+      regRowMaps[l][j]  = Teuchos::rcp_const_cast<Xpetra::Map<LO,GO,NO> >(regMatrices[l][j]->getRowMap()); // ToDo (mayr.mt) Should we rather copy?
```

Comment:
```suggestion
      regRowMaps[l][j]  = Teuchos::rcp_const_cast<Xpetra::Map<LO,GO,NO> >(regMatrices[l][j]->getRowMap());
```

Reply:
We can get rid of this comment, as we now know, that we don't have to copy. Other than that, this change here (and in the next line) is just a change in white spaces.